prover/witness_generator/src/basic_circuits.rs

```diff
@@ -661,17 +703,50 @@ async fn generate_witness(
         }
     };
 
+    let mut eip_4844_blobs = eip_4844_blobs.blobs;
+
+    let mut eip_4844_circuits = vec![];
+    let mut eip_4844_witnesses = vec![];
+
+    let single_blob = eip_4844_blobs.len() == 1;
+    if single_blob {
+        eip_4844_blobs.push(eip_4844_blobs[0].clone());
```

Comment:
wait, what? If we only have one blob, we clone it and make it appear that it's two blobs? If so let's at least add a comment